tests/integration-tests/tests/storage/kms_key_factory.py

```diff
@@ -0,0 +1,207 @@
+import json
+import logging
+import random
+import string
+import time
+
+import boto3
+import pkg_resources
+from jinja2 import Environment, FileSystemLoader
+
+
+class KMSKeyFactory:
+    """Manage creation for kms key."""
+
+    def __init__(self):
+        self.iam_client = None
+        self.kms_client = None
+        self.kms_key_id = None
+        self.account_id = boto3.client("sts").get_caller_identity().get("Account")
+        self.region = boto3.session.Session().region_name
+        self.iam_role = None
+        self.iam_policy_arn = None
+
+    def create_kms_key(self, scheduler):
+        """
+        Create a kms key given scheduler.
+        :param scheduler: Different scheduler need to attach different instance policies
+        """
+        # Only one iam role and one KMS key creation per factory allowed
+        if self.iam_role:
+            raise Exception("IAM Role already created")
+        if self.kms_key_id:
+            raise Exception("KMS key already created")
+
+        self.iam_role = self.create_role(scheduler)
+        self.kms_key_id = self._create_kms_key(scheduler)
+        return self.kms_key_id
+
+    def create_role(self, scheduler):
+        """
+        Create ism role in given scheduler.
+        :param scheduler: Create different roles based on different scheduler
+        """
+        if scheduler == "awsbatch":
+            iam_role_name = "Integration_test_ParallelClusterInstanceRole_batch"
+            random_string = "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(8))
+            iam_policy_name = "".join("Integ_test_ParallelClusterInstancePolicy_batch" + random_string)
+            logging.info("iam_policy_name is {0}".format(iam_policy_name))
+        else:
+            iam_role_name = "Integration_test_ParallelClusterInstanceRole"
+            random_string = "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(8))
+            iam_policy_name = "".join("Integ_test_ParallelClusterInstancePolicy" + random_string)
+
+        self.iam_client = boto3.client("iam")
+
+        # if the role already exist in the account, use the existing one
+        roles = self.iam_client.list_roles().get("Roles")
```

Comment:
This `create_role ` function is getting a bit long.
We could for example extract this part about checking for existing iam role into a separate function called `check_existing_role`

Reply:
Good idea, after our discussion, we create the iam role and destroy it after the test, so we also remove this part of logic now since we don't need to check the existing role